File: inventoryMicroservice/src/main/java/com/example/inventorymicroservice/shared/infrastructure/adapters/inputs/rest/dtos/responses/BaseResponse.java
package com.example.inventorymicroservice.shared.infrastructure.adapters.inputs.rest.dtos.responses;

import lombok.Builder;
import lombok.Getter;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

@Builder
@Getter
public class BaseResponse {
    private Object data;
    private String message;
    private boolean success;
    private HttpStatus httpStatus;

    public ResponseEntity<BaseResponse> toResponseEntity() {
        return new ResponseEntity<>(this, httpStatus);
    }
}

File: ordersMicroservice/src/orders/infrastructure/adapters/inputs/models/baseResponse.ts
export class BaseResponse {
  data: object;
  message: string;
  success: boolean;
  httpStatus: number;

  constructor(data: any, message: string, success: boolean, httpStatus: number) {
    this.data = data;
    this.message = message;
    this.success = success;
    this.httpStatus = httpStatus;
  }
}